Sources/CardPayments/CardClient.swift

```diff
@@ -61,6 +61,7 @@ public class CardClient: NSObject {
                 analyticsService?.sendEvent("card-payments:3ds:confirm-payment-source:failed")
                 notifyFailure(with: error)
             } catch {
+                // TODO: notifyFailure and analytics here?
```

Comment:
I also saw in the CardClient that we are not dealing with all errors like decoding errors. In approveOrder function, we don't send notifyFailure or analytic event in the general catch block . Is there a reason for this?

Reply:
Can you include links to the lines of code in CardClient you're referring to? It is hard to know what part you mean without a reference

Reply:
We should definitely handle the general `catch` case. I don't think we need two `catch` blocks either, it can all just be the general catch block, something like:

```
} catch let error {
    analyticsService?.sendEvent("card-payments:3ds:confirm-payment-source:failed")
    notifyFailure(with: error)
}
```

If you're going to leave a TODO, make sure it is actionable and clear and ideally also has a JIRA ticket number. I personally would remove the TODO for now and just make a JIRA with links to address this in the future.

Reply:
We don't even let the let error (catch defines that property for us for free! 👯 ) can be even further simplified to:
```swift
} catch {
    analyticsService?.sendEvent("card-payments:3ds:confirm-payment-source:failed")
    notifyFailure(with: error)
}
```

Reply:
Compiler is not happy with out the error on notify line.

Reply:
Yeah, I have to change expected error type on notifyFailure and didFinishWith Error.